models/skills.js

```diff
@@ -0,0 +1,72 @@
+const firestore = require("../utils/firestore");
+const skillsModel = firestore.collection("skills");
+const users = require("./users");
+
+/**
+ * Award a Skill
+ *
+ * @param skillData { Object }:  Skill data object to be stored in DB
+ * @param userName  { String }:   Username of user to fetch their details.
+ * @return {Promise<skill|Object>}
+ */
+async function awardSkill(skillData, userName) {
+  try {
+    const skillSnapshot = await skillsModel.add(skillData);
+    const userInfo = await users.fetchUser({ username: userName });
```

Comment:
What happens when the user is not found?

Reply:
There's a 404 response for that with an error and a message 
For e.g.
`
{ 
   "statusCode": 404, 
   "error": "Not Found", 
   "message": "User doesn't exist" 
}
`